src/component/container/LayerTreeClassic/LayerTreeClassic.tsx

```diff
@@ -24,56 +25,44 @@ interface DefaultLayerTreeClassicProps {
   dispatch: (arg: any) => void;
   showContextMenu?: boolean;
   showZoomToLayerExtent?: boolean;
+  showZoomToLayerResolution?: boolean;
   showApplyTimeInterval: boolean;
 }
 
-interface LayerTreeClassicProps extends Partial<DefaultLayerTreeClassicProps> {
-  map: any;
-  t: (arg: string) => any;
+interface LayerTreeClassicProps {
+  map: OlMap;
+  t: (arg: string) => string;
   treeNodeFilter?: (value: any, index: number, array: any[]) => boolean;
 }
 
+type ComponentProps = DefaultLayerTreeClassicProps & LayerTreeClassicProps;
+
 /**
  * Class representing a classic LayerTree with included Legend.
  *
  * @class LayerTreeClassic
- * @extends React.Component
  */
-export class LayerTreeClassic extends React.Component<LayerTreeClassicProps> {
-
-  public static defaultProps: DefaultLayerTreeClassicProps = {
-    extraLegendParams: {
-      LEGEND_OPTIONS: 'fontAntiAliasing:true;forceLabels:on;fontName:DejaVu Sans Condensed',
-      TRANSPARENT: true
-    },
-    dispatch: () => {},
-    showApplyTimeInterval: true,
-    showZoomToLayerExtent: true
-  };
+export const LayerTreeClassic: React.FC<ComponentProps> = ({
+  extraLegendParams = {
+    LEGEND_OPTIONS: 'fontAntiAliasing:true;forceLabels:on;fontName:DejaVu Sans Condensed',
+    TRANSPARENT: true
+  },
+  dispatch,
+  showContextMenu,
+  showApplyTimeInterval = false,
+  showZoomToLayerExtent = false,
```

Comment:
Please note, that this might have an impact on project applications.

Reply:
I've changed the MR description and will add a appropriate note to the CHANGELOG.md later.